pkg/ccl/backupccl/backup_test.go

```diff
@@ -283,6 +283,25 @@ func TestBackupRestoreStatementResult(t *testing.T) {
 	}
 }
 
+func TestBackupRestoreSingleUserfile(t *testing.T) {
```

Comment:
The reason I marked this as a WIP is that I need some help understanding why the new userfile Backup and Restore test behaves the way it does.
- Testing locally, I am able to BACKUP/RESTORE tables and databases correctly (no error thrown).
- The commented out implementation of this test also does not raise any errors.

When I switch to using some of the more involved infrastructure we have for testing the BACKUP/RESTORE of database data, I get a  `pq: importing span /Table/53/1{-/100}: checksum mismatch for 586866600006090753.sst` when the `RESTORE` stmt is runs. All the BACKUP verification steps before it run fine. I'm wondering whether it is something to do with how userfile chunks its data, I'm not sure how to debug this further?

Reply:
I also realized that if I set the splits to 0 when generating the bank data in `backupRestoreTestSetupWithParams` and remove the index created on data.bank, the restore goes through fine. I'm not sure how all these things tie together to cause that checksum mismatch.

Reply:
I looked into this a bit and I think it looks like it has to do with the userfile being able to handle retries. I think the fix would look something like https://github.com/cockroachdb/cockroach/pull/53940.

Reply:
Thanks for looking into this Paul, I'm wondering if it is better to just eliminate the possibility of a txn retry in WriteFile and switch to using the internal executor to just `BEGIN, COMMIT/ROLLBACK`.

This also simplifies the interaction with the copy machine which wraps an `io.PipeReader` in a `noopSeeker` currently. Implementing seek on a pipe reader where we don't have access to all the data at once seems complicated? We would need some buffering/state logic. @miretskiy do you have any inputs here?

Reply:
Hmm, I think that the retry semantics are useful since the transaction may get a retry request. But it looks like only the creation of writer needs to be transactional, so it looks like we can move the writing of the data outside the txn?